### ISSUE TITLE
@usecase crashes scaladoc 2.13.0 with ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by nafg** [scala/bug#11572]
Compiling this https://github.com/nafg/slick-migration-api/blob/aefad025ad4cc2371d97544bd8457852b66988e6/src/main/scala/slick/migration/api/Migration.scala#L18 for 2.13.0 causes scaladoc to crash.

It works fine on 2.12.8.

Here is the output:

```
[warn] /home/naftoli/dev/github.com/nafg/slick-migration-api/src/main/scala/slick/migration/api/Migration.scala:18:17: The @usecase tag is deprecated, instead use the @example tag to document the usage of your API
[warn]      * @usecase def &(n: ReversibleMigration): ReversibleMigrationSeq
[warn]                 ^
[error] java.lang.ArrayIndexOutOfBoundsException: 43
[error]         at scala.reflect.internal.util.BatchSourceFile.findLine$1(SourceFile.scala:200)
[error]         at scala.reflect.internal.util.BatchSourceFile.offsetToLine(SourceFile.scala:203)
[error]         at xsbt.DelegatingReporter$.lineOf$1(DelegatingReporter.scala:99)
[error]         at xsbt.DelegatingReporter$.makePosition$1(DelegatingReporter.scala:112)
[error]         at xsbt.DelegatingReporter$.convert(DelegatingReporter.scala:134)
[error]         at xsbt.DelegatingReporter.info0(DelegatingReporter.scala:165)
[error]         at scala.reflect.internal.Reporter.warning(Reporting.scala:91)
[error]         at scala.tools.nsc.Reporting$PerRunReporting$ConditionalWarning.warn(Reporting.scala:42)
[error]         at scala.tools.nsc.Reporting$PerRunReporting.deprecationWarning(Reporting.scala:79)
[error]         at scala.tools.nsc.ast.DocComments$DocComment.decomposeUseCase(DocComments.scala:415)
[error]         at scala.tools.nsc.ast.DocComments$DocComment.$anonfun$x$15$5(DocComments.scala:403)
[error]         at scala.tools.nsc.ast.DocComments$DocComment.x$15$lzycompute(DocComments.scala:403)
[error]         at scala.tools.nsc.ast.DocComments$DocComment.x$15(DocComments.scala:393)
[error]         at scala.tools.nsc.ast.DocComments$DocComment.defines$lzycompute(DocComments.scala:393)
[error]         at scala.tools.nsc.ast.DocComments$DocComment.defines(DocComments.scala:393)
[error]         at scala.tools.nsc.ast.DocComments$DocComment.defineVariables(DocComments.scala:431)
[error]         at scala.tools.nsc.doc.ScaladocAnalyzer$ScaladocTyper.typedDocDef(ScaladocAnalyzer.scala:42)
[error]         at scala.tools.nsc.doc.ScaladocAnalyzer$ScaladocTyper.typedDocDef$(ScaladocAnalyzer.scala:36)
[error]         at scala.tools.nsc.doc.ScaladocAnalyzer$$anon$1.typedDocDef(ScaladocAnalyzer.scala:25)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typedOutsidePatternMode$1(Typers.scala:5813)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5831)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5867)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:5931)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$10(Typers.scala:3396)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3396)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typedTemplate(Typers.scala:2050)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typedClassDef(Typers.scala:1880)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5775)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5867)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:5931)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$10(Typers.scala:3396)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3396)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typedTemplate(Typers.scala:2050)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typedModuleDef(Typers.scala:1925)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5776)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5867)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:5931)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$10(Typers.scala:3396)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3396)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typedPackageDef$1(Typers.scala:5475)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5778)
[error]         at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:5867)
[error]         at scala.tools.nsc.typechecker.Analyzer$typerFactory$TyperPhase.apply(Analyzer.scala:115)
[error]         at scala.tools.nsc.Global$GlobalPhase.applyPhase(Global.scala:451)
[error]         at scala.tools.nsc.typechecker.Analyzer$typerFactory$TyperPhase.run(Analyzer.scala:104)
[error]         at scala.tools.nsc.Global$Run.compileUnitsInternal(Global.scala:1510)
[error]         at scala.tools.nsc.Global$Run.compileUnits(Global.scala:1494)
[error]         at scala.tools.nsc.Global$Run.compileSources(Global.scala:1486)
[error]         at scala.tools.nsc.Global$Run.compile(Global.scala:1615)
[error]         at scala.tools.nsc.doc.DocFactory.makeUniverse(DocFactory.scala:51)
[error]         at scala.tools.nsc.doc.DocFactory.generate$1(DocFactory.scala:131)
[error]         at scala.tools.nsc.doc.DocFactory.document(DocFactory.scala:138)
[error]         at xsbt.Runner.run(ScaladocInterface.scala:34)
[error]         at xsbt.ScaladocInterface.run(ScaladocInterface.scala:19)
[error]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error]         at java.base/java.lang.reflect.Method.invoke(Method.java:564)
[error]         at sbt.internal.inc.AnalyzingCompiler.call(AnalyzingCompiler.scala:248)
[error]         at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:177)
[error]         at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:158)
[error]         at sbt.Doc$.$anonfun$scaladoc$1(Doc.scala:39)
[error]         at sbt.Doc$.$anonfun$scaladoc$1$adapted(Doc.scala:39)
[error]         at sbt.RawCompileLike$.$anonfun$prepare$1(RawCompileLike.scala:83)
[error]         at sbt.RawCompileLike$.$anonfun$prepare$1$adapted(RawCompileLike.scala:76)
[error]         at sbt.RawCompileLike$.$anonfun$cached$3(RawCompileLike.scala:67)
[error]         at sbt.RawCompileLike$.$anonfun$cached$3$adapted(RawCompileLike.scala:65)
[error]         at sbt.util.Tracked$.$anonfun$inputChanged$1(Tracked.scala:150)
[error]         at sbt.RawCompileLike$.$anonfun$cached$1(RawCompileLike.scala:72)
[error]         at sbt.RawCompileLike$.$anonfun$cached$1$adapted(RawCompileLike.scala:55)
[error]         at sbt.Defaults$.$anonfun$docTaskSettings$3(Defaults.scala:1549)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:67)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:280)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
[error]         at sbt.Execute.work(Execute.scala:289)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:280)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:514)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1135)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[error]         at java.base/java.lang.Thread.run(Thread.java:844)
```

**Comment by SethTisue** [comment]
you mentioned on Gitter that multiple `@usecase` annotations were involved, is that crucial to reproducing the problem?

**Comment by nafg** [comment]
The invalid index always seems to be the last line of the file, so sounds like an off-by-one. However sometimes I think I was able to fix it by adding blank lines at the end of the file.

The following minimization seems to reproduce it.

```scala
object Test {
  /**
   * @usecase def x(n: Any): Any
   */
  def x(n: Any): Any = null
}
```

**Comment by hamednourhani** [comment]
@nafg can I work on it?

**Comment by nafg** [comment]
You have my full permission!

**Comment by xuwei-k** [comment]
https://github.com/sbt/sbt/issues/4665 🤔

**Comment by nafg** [comment]
Interesting, that mentions https://github.com/scala/bug/issues/9885, which was fixed, but nevertheless I should note that my editor always saves with a trailing newline.

**Comment by xuwei-k** [comment]
I think this is a sbt bug not scalac bug🤔
/cc @eed3si9n

**Comment by xuwei-k** [comment]
see
- https://github.com/sbt/sbt/issues/4665#issuecomment-507014877
- https://github.com/xuwei-k/sbt-usecase-bug

**Comment by nafg** [comment]
@xuwei-k thanks for doing all that digging, but it doesn't match what I saw.

In my case, it works fine on sbt 1.2.8 and scala 2.12.8. It only crashed on scala 2.13.0.

**Comment by xuwei-k** [comment]
Hmmm 🤔 OK, reopen this issue for now.

**Comment by som-snytt** [comment]
Duplicates https://github.com/scala/bug/issues/11865 which I went to a great deal of trouble today to fix again. But donated a test in the linked PR.

**Comment by som-snytt** [comment]
Worth noting the fix is on 2.12.11/2.13.2 but only certain versions of sbt/zinc exploit the bug.
